com.unity.netcode.gameobjects/Components/Interpolator/BufferedLinearInterpolator.cs

```diff
@@ -151,7 +151,11 @@ public T Update(float deltaTime)
 
                 if (Debug.isDebugBuild)
                 {
-                    Debug.Assert(t >= 0, $"t must be bigger than or equal to 0. range {range}, RenderTime {RenderTime}, Start time {m_StartTimeConsumed.Time}, end time {m_EndTimeConsumed.Time}");
+                    if (t < 0)
+                    {
+                        Debug.LogWarning(
```

Comment:
I'd still assert here, this is not something normal

Reply:
If by 'assert' you mean terminate, TIL Debug.Assert doesn't do that.  It just does an assert-level log.

Reply:
I would just use LogError instead of warning because I think we do want it to be an error?

Reply:
I synced with Matt in private chat, to me, this should be an Exception, we should interrupt the flow and fail early

Reply:
the RenderTime is controlled by the interpolator. It's based on the tick system's server time, which doesn't move linearly and can be adjusted. The situation where we have a negative t shouldn't arise, but better be safe. I prefer to fail violently than silently

Reply:
Ok, I still find this a bit inconsistent.  Also am not thrilled we allow for division by potentially very small numbers so long as they're not zero.